instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -110,99 +102,69 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
+                .first()
+            mapboxNavigation.roadObjectsOnRoute()
+                .filter { upcomingRoadObjects ->
+                    upcomingRoadObjects.size == 2 &&
+                        upcomingRoadObjects.map { it.roadObject.id }
+                            .containsAll(listOf("11589180127444257", "14158569638505033"))
+                }
+                .first()
 
-        // Wait for the route refresh.
-        val refreshedRoutes = initialRouteIdlingResource.next()
-        initialRouteIdlingResource.unregister()
+            assertEquals(
+                listOf("11589180127444257"),
+                initialRoutes[0].getIncidentsIdFromTheRoute(0)
+            )
+            assertEquals(
+                listOf("11589180127444257", "14158569638505033").sorted(),
+                refreshedRoutes[0].getIncidentsIdFromTheRoute(0)?.sorted()
+            )
+            assertEquals(
+                listOf("11589180127444257"),
+                initialRoutes[1].getIncidentsIdFromTheRoute(0)
+            )
+            assertEquals(
+                listOf("11589180127444257", "14158569638505033").sorted(),
+                refreshedRoutes[1].getIncidentsIdFromTheRoute(0)?.sorted()
+            )
 
-        val progressIdlingResource = RouteProgressStateIdlingResource(
-            mapboxNavigation,
-            RouteProgressState.TRACKING
-        )
-        progressIdlingResource.register()
-        Espresso.onIdle()
-        progressIdlingResource.unregister()
+            assertEquals(
+                requestedRoutes[0].getDurationOfLeg(0),
+                initialRoutes[0].getDurationOfLeg(0),
+                0.0
+            )
+            assertEquals(227.918, initialRoutes[0].getDurationOfLeg(0), 0.0001)
+            assertEquals(1189.651, refreshedRoutes[0].getDurationOfLeg(0), 0.0001)
 
-        val latch = CountDownLatch(2)
-        runOnMainSync {
-            mapboxNavigation.registerRouteProgressObserver { routeProgress ->
-                if (isRefreshedRouteDistance(routeProgress)) {
-                    latch.countDown()
-                }
-            }
-            mapboxNavigation.registerRoadObjectsOnRouteObserver { upcomingRoadObjects ->
-                // upcoming road's objects mut be exact 2 with following ids
-                if (upcomingRoadObjects.size == 2 &&
-                    upcomingRoadObjects.map { it.roadObject.id }
-                        .containsAll(listOf("11589180127444257", "14158569638505033"))
-                ) {
-                    latch.countDown()
-                }
-            }
-        }
-        if (!latch.await(10, TimeUnit.SECONDS)) {
-            throw AssertionError(
-                """progress duration remaining or upcoming road objects weren't  
-                    refreshed by native navigator
-                """.trimMargin()
+            assertEquals(
+                requestedRoutes[1].getDurationOfLeg(0),
```

Comment:
should be enough to check `annotation().duration()`? The test also show that items are not re-arranged and it's actually the same data

Reply:
updated